src/common/snippets/src/pass/transpose_decomposition.cpp

```diff
@@ -39,38 +40,22 @@ ngraph::snippets::pass::TransposeDecomposition::TransposeDecomposition() {
         if (supported_cases.count(order_value) == 0)
             return false;
 
-        auto data_input = pattern_to_output.at(match_data);
-        const auto& data_node = pattern_to_output.at(match_data).get_node_shared_ptr();
-        auto &param_rt = data_node->get_rt_info();
-        // Note: store and usage inside emitters as size_t is more convenient, so static_cast here
-        const auto& access_pattern = order->cast_vector<size_t>();
-        param_rt["Layout"] = access_pattern;
+        // number of elements that can be processed on every iteration. For 0,1,2,3 -> 0,2,3,1 we can guarantee only scalar access
+        const auto subtensor = std::vector<size_t>{1};
+        const auto& layout = order->cast_vector<size_t>();
 
-        // The line below is Ok, since we ensured that transpose is static above
-        auto data_shape = data_input.get_shape();
-        // dim indexes with respect to SRC
-        const auto dim_C_idx = data_shape.size() - 3;
-        const auto dim_H_idx = data_shape.size() - 2;
-        const auto dim_W_idx = data_shape.size() - 1;
-        const auto size_C = static_cast<int64_t>(data_shape[dim_C_idx]);
-        const auto size_W = static_cast<int64_t>(data_shape[dim_W_idx]);
-        const auto size_H = static_cast<int64_t>(data_shape[dim_H_idx]);
-
-        auto loop_W_begin = std::make_shared<op::LoopBegin>(OutputVector{data_input});
-        auto loop_C_begin = std::make_shared<op::LoopBegin>(OutputVector{loop_W_begin->output(0)});
         // todo: LoadReshape used here is essentially Load + an easy way to maintain correct shape propagation
         //  fix this in future and develop a more consistent shape propagation approach.
-        auto load = std::make_shared<snippets::op::LoadReshape>(loop_C_begin->output(0), 1, 0, access_pattern);
-        auto store = std::make_shared<snippets::op::Store>(load, 1);
-        const std::vector<int64_t> ptr_increments_C {size_H * size_W, 1};
-        const std::vector<int64_t> finalization_offsets_C {1 - size_H * size_W * size_C, 0};
-        auto loop_C_end = std::make_shared<op::LoopEnd>(OutputVector{store->output(0), loop_C_begin->output(1)},
-                                                        size_C, 1, ptr_increments_C, finalization_offsets_C);
-        auto loop_W_end = std::make_shared<op::LoopEnd>(OutputVector{loop_C_end->output(0), loop_W_begin->output(1)},
-                                                        size_W, 1, std::vector<int64_t>{0, 0}, std::vector<int64_t>{0, 0});
+        auto load = std::make_shared<snippets::op::LoadReshape>(data_input, subtensor[0], 0, layout);
+        auto store = std::make_shared<snippets::op::Store>(load, subtensor[0]);
+
+        lowered::PortManager::set_port_descriptor_ptr(load->input(0), std::make_shared<lowered::PortDescriptor>(load->get_input_shape(0), subtensor, layout));
```

Comment:
@IvanNovoselov  BTW, `PortDescriptor` is incapsulated into ngraph op semantics. So why it is part of `lowered` namespace?

Reply:
`PortDescriptor` is in the `lowered` namespace because it's primarily used in the LIR pipeline. An ngraph op can save a `PortDescriptor` into the node's rt_info, but this is rarely needed. So it's more like we allow a node to provide some low-level lowering info, rather than we use ngraph entity in the LIR pipeline.